src/app/Library/CrudPanel/Traits/Buttons.php

```diff
@@ -241,20 +249,31 @@ public function getButtonKey($name)
         $array = $this->buttons()->toArray();
 
         foreach ($array as $key => $value) {
-            if ($value->name == $name) {
+            if ((is_object($value) ? $value->name : $value['name']) == $name) {
                 return $key;
             }
         }
     }
 
+    /**
+     * Return the buttons for a given stack.
+     *
+     * @param  string  $stack
+     * @return \Illuminate\Support\Collection
+     */
+    public function getButtonsForStack(string $stack)
+    {
+        return $this->buttons()->where('stack', $stack);
+    }
+
     /**
      * Add a new button to the current CRUD operation.
      *
      * @param  string|array  $attributes  Button name or array that contains name, stack, type and content.
      * @return \Backpack\CRUD\app\Library\CrudPanel\CrudButton
      */
-    public function button($attributes = null)
+    public function button(string|array $name)
```

Comment:
If in this variable we accept two things, why not say that in the variable name? That would completely explain it, right?

![why-dont-we-have-both-gif](https://media.tenor.com/cHYRjCIXyckAAAAC/why-not-both-por-que-no-los-dos.gif)


```suggestion
    public function button(string|array $nameOrAttributesArray)
```